modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/index/AbstractSchemaSelfTest.java

```diff
@@ -268,6 +272,39 @@ static void assertIndex(Ignite node, String cacheName, String tblName,
         assertInternalIndexParams(typeDesc, idxName, inlineSize);
     }
 
+    /**
+     * Forms full list of index fields. It is assumed that cache has the only QueryEntity.
+     *
+     * @param cache Cache with index.
+     * @param idxName Index name.
+     * @param fields Index fields specified in configuration or DDL.
+     */
+    static Collection<IgniteBiTuple<String, Boolean>> idxFields(IgniteCache<?, ?> cache, String idxName,
+        IgniteBiTuple<String, Boolean>... fields) {
+        QueryEntity qryEntity = (QueryEntity)cache.getConfiguration(CacheConfiguration.class)
+            .getQueryEntities().iterator().next();
+
+        boolean isStaticIdx = qryEntity.getIndexes().stream()
```

Comment:
We should not rely on that, it actually looks like a bug. QueryEntity must be up to date after index creation.

Reply:
I agree, I will fix this and create an ticket for investigation.